packages/coreutils/src/interfaces.ts

```diff
@@ -37,10 +37,11 @@ interface IDataConnector<T, U> {
    * @returns A promise that bears a data payload if available.
    *
    * #### Notes
-   * The promise returned by this method may be rejected if an error occurs in
-   * retrieving the data. Non-existence of an `id` will succeed with `null`.
+   * The promise returned by this method may be rejected if an error
+   * occurs in  retrieving the data. Non-existence of an `id` will
+   * succeed with `undefined`.
    */
-  fetch(id: string): Promise<T | null>;
+  fetch(id: string): Promise<T | undefined>;
```

Comment:
I think we need to update the `StateDB` to implement these interface changes. They aren't being caught right now because we aren't compiling with strict null checks. I think we need to turn on that flag ASAP.